test/framework/README.md

```diff
@@ -0,0 +1,38 @@
+# CAPI e2e testing framework
+
+This framework aims to define common end-to-end patterns that can be reused across Cluster API providers.
+
+## Usage
+
+Use this framework as you see fit. If there are pieces that don't work for you, do not use them. If there are pieces
+that almost work for you, customize them. If this does not fit your e2e testing please file an issue and we can discuss
+your use case and find a nice solution for everyone.
+
+### Contents
+
+This framework contains
+
+* An interface to a management cluster
+    * An struct that implements the management cluster interface using `kind` as the backend.
+* A series of behavioral tests
+* Clean up tests.
+
+## Requirements
```

Comment:
Sounds like `kind` is going to be a requirement?

Reply:
No, kind should not be a requirement. We provide helpers to set up kind, but so long as you have some code that wraps a kubernetes cluster to satisfy the interface provided, kind is not a requirement.